src/mlpack/methods/ann/layer/softshrink.hpp

```diff
@@ -0,0 +1,224 @@
+/**
+ * @file softshrink.hpp
+ * @author Lakshya Ojha
+ *
+ * The soft shrink function has thresholds proportional to the noise level given by 
+ * the user.
+ * The use of a soft-shrinkage activation function provides adaptive denoising
+ * at various noise levels using a single CNN(Convolution Neural) without a requirement to train a
+ * unique CNN for each noise level.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SOFTSHRINK_HPP
+#define MLPACK_METHODS_ANN_LAYER_SOFTSHRINK_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artifical Neural Network. */ {
+
+/**
+ * SoftShrinkage operator is defined as,
+ *         | x - lambda, if x >  lambda
```

Comment:
I guess it would be nice to use the doxygen syntax here which will create a neat image.

Reply:
Hi @zoq 
According to what I saw about doxygen syntax I got this idea, as I am not sure if this is correct.
Kindly help me with the same if it needs changes.
Thanks.
![Screenshot from 2020-02-19 17-59-35](https://user-images.githubusercontent.com/57477999/74834554-a554d000-5341-11ea-8b6b-e8a2897428a7.png)

Reply:
Looks like you figured it out.